### ISSUE TITLE
getProperties().get returns an ODocument, while getProperty returns an OrientVertex

### ISSUE THREAD

**Issue by benjaminbillet** [orientechnologies/orientdb#6719]
## OrientDB Version, operating system, or hardware.

v2.2.10, MacOS X
## Expected behavior and actual behavior

I was testing the result of a MATCH request and I noticed that the result vertices are returned either as ODocument or OrientVertex depending on the method used:
- OrientVertex#getProperties returns a map where the vertices are of type ODocument
- OrientVertex#getProperty returns vertices as OrientVertex.

Is this the expected behavior?

Here is some code for testing the issue:

``` java
String query = "SELECT FROM (MATCH {class: TypeA, as: data, where: (@rid = '" + root.getId() + "')} RETURN $paths)";

Iterable<OrientVertex> rows = graph.command(new OCommandSQL(query)).execute();
OrientVertex row = rows.iterator().next();

System.out.println(row.getProperties().get("data").getClass()); 
System.out.println(row.getProperty("data").getClass());
// output:
// class com.orientechnologies.orient.core.record.impl.ODocument
// class com.tinkerpop.blueprints.impls.orient.OrientVertex
```

The result of the request in the console appears to be okay (a regular rid):

```
>SELECT FROM (MATCH {class: TypeA, as: data, where: (@rid = '#89:18')} RETURN $paths)
+----+------+
|#   |data  |
+----+------+
|0   |#89:18|
+----+------+
```

**Comment by luigidellaquila** [comment]
Hi @benjaminbillet 

Thank you for reporting, for sure this can be considered an inconsistency. 
I'll check if it is possible to change this behavior in v 2.2.x hotfix, as it will change current behavior.
For sure it will change in next major/minor release

Thanks

Luigi

**Comment by tglman** [comment]
hi @benjaminbillet,

for not breaking compatibility with existing applications we are going to schedule this change for 3.0

Bye

**Comment by benjaminbillet** [comment]
That's fine. This inconsistency is not highly blocking :)
Thanks.

**Comment by lvca** [comment]
@tglman is this already fixed in 3.0?

**Comment by tglman** [comment]
hi,

This is not fixed in 3.0 but TP2 has been deprecated and TP 3.0 or core 3.0 provide a different API.

So this can be closed.

Regards
